packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1396,13 +1396,13 @@ abstract class RenderSliver extends RenderObject {
   /// of the sliver whereas [childScrollOffset] gives the distance from the
   /// sliver's zero scroll offset.
   ///
-  /// Calling this for a child that is not visible is not valid.
+  /// Calling this for a child that is not visible should return null.
   @protected
   double childMainAxisPosition(covariant RenderObject child) {
     assert(() {
       throw FlutterError('${objectRuntimeType(this, 'RenderSliver')} does not implement childPosition.');
     }());
-    return 0.0;
+    return null;
```

Comment:
This isn't a safe change; most callers assume that the return-value is non-null.

Reply:
The other way will be somehow prevent everything below the kept alive child from being paint. Let me scratch my head a bit...